pkg/columns/columninfo.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package columns
```

Comment:
I don't know, the "higher" package shouldn't be also the "higher" level of the API? IMO, even if the users call also `columns` directly, the higher level of the API is `textcolumns`, isn't it? In fact, `textcolumns` uses `columns` and not the other way around. Therefore, `columns` should be a sub-package of `textcolumns`. Don't you think?

Reply:
The plan was to handle everything aside from the actual tag parsing and preparation of the `Columns` in different feature packages to make it as independent as possible. So every feature relies on the spec from the `Columns` and uses it, yet is completely independent from all other features. An example where it should be used:
We pre-sort things on the gadget side with the top gadgets because we only return a limited number of rows when there are too many. There you would use the sorter (and maybe the filter), but not the formatter (cause that's only used in the "UI" part).

Reply:
Makes sense!